guides/specification.md

```diff
@@ -0,0 +1,549 @@
+# Trento Checks DSL Specification
+
+A language allowing to declare best practices to be adhered on target SAP Infrastructures.
+
+# Table of contents
+
+- [Introduction](#introduction)
+- [Anatomy of a Check](#anatomy-of-a-check)
+  - [Filename Convention](#filename-convention)
+  - [Structure](#structure)
+    - [id](#id)
+    - [name](#name)
+    - [group](#group)
+    - [description](#description)
+    - [remediation](#remediation)
+    - [severity](#severity)
+    - [Facts, Values, Expectations](#facts-values-expectations)
+- [Facts](#facts)
+- [Values](#values)
+- [Expectations](#expectations)
+- [Expression Language](#expression-language)
+    - [Evaluation Scope](#evaluation-scope)
+
+## Introduction
+
+The need this DSL aims to fulfill is to provide users a simple way to declare what we (the Trento Team) often refer to as "Checks". 
+
+Checks are, in Trento's domain, the crystallization of SUSE's best practices when it comes to SAP workloads in a form that both a user and a machine can read.
+
+This approach provides several benefits:
+
+- Users can formalize best practices with no space for ambiguity;
+- Machines can assert systems' state, automatically, with no space for ambiguity;
+- The development of new best practices gets streamlined through a common definition that allows to firestart shared efforts.
```

Comment:
This intro does well in an RFC/ADR/design document or marketing document.
I think it's too fuzzy for documentation where we want to be more specific on what the application does, I would rephrase it in simpler terms without stating the benefits.
The focus should be on how it works rather than why we did it.

Also, there is no high-level explanation of how the check flow works.
E.g. execution requested -> facts gathered -> expectation evaluated.
I think this is really important for whoever implements a check to understand the checks lifetime.

Reply:
Changed here 4af3b8dab590c427e4938491c842a4d300d9b1fa